apps/coscrad-frontend/src/components/higher-order-components/aggregate-page.tsx

```diff
@@ -0,0 +1,154 @@
+import { buildStaticCommandExecutors } from '../commands/build-static-command-executors';
+import { buildCommandExecutor, buildDynamicCommandForm } from '../commands/command-executor';
+import { ErrorDisplay } from '../error-display/error-display';
+import { Loading } from '../loading';
+import { NoteDetailPageContainer } from '../notes/note-detail-page.container';
+import { buildUseLoadableSearchResult } from './buildUseLoadableSearchResult';
+
+type DetailPresenter = (viewModel: IDetailQueryResult<IBaseViewModel>) => JSX.Element;
+
+interface ResourcePageProps {
+    aggregateType: AggregateType;
+    detailPresenterFactory: (categorizableType: AggregateType) => DetailPresenter;
+}
+
+const isCategorizableCompositeIdentifier = (
+    input: unknown
+): input is CategorizableCompositeIdentifier => {
+    if (input === null || typeof input === 'undefined') return false;
+
+    const { type } = input as CategorizableCompositeIdentifier;
+
+    return Object.values(CategorizableType).includes(type);
+};
+
+const buildCommandExecutionFormsAndLabels = (
+    actionsFromApi: IBackendCommandFormAndLabels[],
+    compositeIdentifier: AggregateCompositeIdentifier
+) => {
+    const commandExecutionFormsAndLabels = actionsFromApi
+        .map(
+            (action): ICommandExecutorAndLabels => ({
+                ...action,
+                executor: buildCommandExecutor(buildDynamicCommandForm(action), {
+                    /**
+                     * Any command that naturally appears in the context of
+                     * a categorizable page is an update command for the
+                     * current categorizable (resource or note) in whose detail
+                     * view the user is presently working. Therefore, we must
+                     * bind the `aggregateCompositeIdentifier` prop for this
+                     * categorizable.
+                     */
+                    aggregateCompositeIdentifier: compositeIdentifier,
+                }),
+            })
+        )
+        /**
+         * TODO We should expose the following commands via the back-end.
+         * that will require a mechanism to bind the aggregate context
+         * for the resource we are viewing (e.g. book/123) to a non-standard
+         * property (resourceCompositeIdentifier) while generating a new
+         * ID for the note, which is the proper aggregate context for
+         * this command.
+         */
+        .concat(buildStaticCommandExecutors(compositeIdentifier));
+
+    return commandExecutionFormsAndLabels;
+};
+
+/**
+ * A Categorizable is a Resource or a Note. These types of aggregate root share in
+ * common that they can be tagged and categorized, and they collectively constitute
+ * the web-of-knowledge.
+ *
+ * We may want to extends this to an `AggregatePage`. We aren't too concerned
+ * about that right now as our core value is being extensbile to adding new resource
```

Comment:
(sp) extensible

Reply:
Actually, the entire comment was stale. I have removed it.